src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_emitter.cpp

```diff
@@ -98,136 +106,109 @@ void jit_emitter::emitter_preamble(const std::vector<size_t>& in_idxs,
         OPENVINO_THROW("Failed to allocate required number of gpr registers");
     }
 
+    // vector registers
     for (auto idx : pool_aux_vec_idxs) {
         aux_vec_idxs.push_back(static_cast<uint32_t>(idx));
     }
 
+    for (size_t idx = 0; idx < get_max_vecs_count(); idx++) {
+        if (aux_vec_idxs.size() >= get_aux_vecs_count()) break;
+
+        if (std::find(in_idxs.begin(), in_idxs.end(), idx) != in_idxs.end()) continue;
+        if (std::find(out_idxs.begin(), out_idxs.end(), idx) != out_idxs.end()) continue;
+
+        if (std::find(aux_vec_idxs.begin(), aux_vec_idxs.end(), idx) != aux_vec_idxs.end()) continue;
+        if (std::find(preserved_vec_idxs.begin(), preserved_vec_idxs.end(), idx) != preserved_vec_idxs.end()) continue;
+
+        aux_vec_idxs.push_back(idx);
+        preserved_vec_idxs.push_back(idx);
+    }
+    if (aux_vec_idxs.size() < get_aux_vecs_count())
+        OV_CPU_JIT_EMITTER_THROW("Failed to allocate required number of vector registers");
+
+    // gpr registers
     for (auto idx : pool_aux_gpr_idxs) {
-        aux_gpr_idxs.push_back(static_cast<uint32_t>(idx));
-        preserved_gpr_idxs.push_back(static_cast<uint32_t>(idx));
+        aux_gpr_idxs.push_back(idx);
     }
 
+    const uint32_t end_gpr_idx = Xbyak_aarch64::Operand::X28;
+    for (size_t gpr_idx = 0; gpr_idx <= end_gpr_idx; ++gpr_idx) {
+        size_t _idx = end_gpr_idx - gpr_idx; // we allocate from the end
+
+        if (aux_gpr_idxs.size() >= get_aux_gprs_count()) break;
+        if (_idx == Xbyak_aarch64::Operand::X18) continue;
```

Comment:
Thanks, reworked for `X23(X_TMP_0)`, `X24(X_TMP_1)` and `X28(X_DEFAULT_ADDR)`. As I know OpenVINO arm64 jit emitters don't call oneDNN functions which use `X24(X_TMP_1)`. Can you, please, provide example?

Reply:
I thought `X24(X_TMP_1)` was used here https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/nodes/kernels/aarch64/jit_uni_eltwise_generic.hpp#L101. Now I see it's a different gpr. I am also using this `X24(X_TMP_1)` in arm snippets PR. Anyway, let's do not include the `X24(X_TMP_1)` revision in this pr, since it's not used for now.